misc-scripts/xref_mapping/XrefParser/MGI_Desc_Parser.pm

```diff
@@ -57,59 +56,49 @@ sub run {
-    if ($header ne $header_template) {die "File header has altered from format parser expects. Check MGI "};
-      
-    while ( my $line = $mgi_io->getline() ) {
-
-        chomp($line);
-        my ($accession, $chromosome, $position, $start, $end, $strand,$label, 
-            $status, $marker, $marker_type, $feature_type, $synonym_field) = split(/\t/,$line);
-            
-        $position =~ s/^\s+// if ($position);
-
-        my @synonyms = split(/\|/,$synonym_field) if ($synonym_field);
-        
-	
-        my $desc;
-	if ($marker) {
-	    $desc = $marker;
-	}
-        
-        $acc_to_xref{$accession} = $self->add_xref({ acc        => $accession,
-    	                           		             label      => $label,
-    					                             desc       => $desc,
-    					                             source_id  => $source_id,
-    					                             species_id => $species_id,
-                                                                     dbi        => $dbi,
-    					                             info_type  => "MISC"} );
-        if($verbose and !$desc){
-    	   print "$accession has no description\n";
-        }
-        $xref_count++;
-            
-        if(defined($acc_to_xref{$accession})){
-           
-            foreach my $syn (@synonyms) {
-                $self->add_synonym($acc_to_xref{$accession}, $syn, $dbi);
-                $syn_count++;
-            }
-            
-        }
-        
+  my $input_file = Text::CSV->new({
+                                    sep_char           => "\t",
+				    empty_is_undef     => 1,
+                                    strict => 1,
+                                    auto_diag => 1,
+                                    allow_loose_quotes => 1,
+				   }) or croak "Cannot use file $file: ".Text::CSV->error_diag ();
+
+  # skip header
+  $input_file->getline($mgi_io);
+  $input_file->column_names([qw(accession chromosome position start end strand label status marker marker_type feature_type synonym_field)] );
+  while ( my $data = $input_file->getline_hr( $mgi_io ) ) {
+    my $accession = $data->{'accession'};
+    my $marker = defined($data->{'marker'}) ? $data->{'marker'} : undef;
+    $acc_to_xref{$accession} = $self->add_xref({ acc        => $accession,
+                                                 label      => $data->{'label'},
+                                                 desc       => $marker,
+                                                 source_id  => $source_id,
+                                                 species_id => $species_id,
+                                                 dbi        => $dbi,
+                                                 info_type  => "MISC"} );
+    if($verbose and !$marker){
```

Comment:
Please replace either _and_ with _&&_ (preferred), or _!_ with _not_. Probably harmless in this context but I have already been bitten by the fact the two forms of boolean operators in Perl have got dramatically different priority.